python-lib/utils.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+"""Module with utility functions which are *not* based on the Dataiku API"""
+
+import logging
+import functools
+from typing import Callable, AnyStr
+from time import perf_counter
+
+
+def time_logging(log_message: AnyStr):
+    """Decorator to log timing with a custom message"""
+
+    def inner_function(function: Callable):
+        @functools.wraps(function)
```

Comment:
Could you explain what is the design of this function ?

Reply:
It's a standard timing and logging decorator, it's used as such: https://github.com/dataiku/dss-plugin-similarity-search/blob/a81a6e18a9eebefbfd3c9b275dce70b09a68a3bf/python-lib/nearest_neighbor/annoy.py#L34